imports/plugins/included/discount-codes/server/methods/methods.js

```diff
@@ -32,6 +32,140 @@ export const methods = {
+    let currentDiscount = 0;
+    for (billing of cart.billing) {
+      if (billing.paymentMethod && billing.paymentMethod.processor === "discount-code" && billing._id !== codeId) {
+        currentDiscount += parseFloat(billing.paymentMethod.amount);
+      }
+      if (billing.paymentMethod && billing.invoice) {
+        hasInvoice = true;
+      }
+    }
+    // only if this is an order
+    if (hasInvoice) {
+      selector = {
+        "_id": id,
+        "billing._id": cart.billing[0]._id
+      };
+      update = {
+        $set: {
+          "billing.$.invoice.discounts": currentDiscount
+        }
+      };
+      Collection.update(selector, update);
+    }
+    // TODO: update a history record of transaction
+    // TODO: recalculate cart discounts (not simply 0)
+    return Collection.update(
+      { _id: id },
+      { $set: { discount: currentDiscount }, $pull: { billing: { _id: codeId } } },
+      { multi: true }
+    );
+  },
+  /**
+   * discounts/codes/apply
+   * checks validity of code conditions and then
+   * applies a discount as a paymentMethod to cart
+   * @param  {String} id cart id of which to remove a code
+   * @param  {String} code valid discount code
+   * @param  {String} collection collection (either Orders or Cart)
+   * @return {Boolean} returns true if successfully applied
+   */
+  "discounts/codes/apply": function (id, code, collection = "Cart") {
+    check(id, String);
+    check(code, String);
+    check(collection, String);
+
+    // TODO: further expand to meet all condition rules
+    // const conditions = {
+    //   enabled: true
+    // };
+
+    // TODO: add  conditions: conditions
+    const discount = Discounts.findOne({ code: code });
+
+    // TODO: check usage limit
+    // don't apply if cart has exceeded usage limit
+    // will also need to check all time usage.
+    // which means storing the use data with the Discounts
+    // or searching all user's order history
+    // and if a user cancels an order,
+    // is the discount now re-activated
+
+
+    if (discount) {
+      // save to payment methods
+      // and update status in Discounts
+      // payment methods can be debit or credit.
+      const paymentMethod = {
+        processor: "discount-code",
+        method: "debit",
+        code: discount.code,
+        transactionId: Random.id(),
+        amount: discount.discount, // pre-process to amount.
+        status: "created"
+      };
+      // apply to cart / order
+      Meteor.call("payments/apply", id, paymentMethod, collection, (error, result) => {
+        if (result) {
+          const Collection = Reaction.Collections[collection];
+          const cart = Collection.findOne(id);
+          let hasInvoice = false;
+          let currentDiscount = 0;
+          for (billing of cart.billing) {
+            if (billing.paymentMethod && billing.paymentMethod.processor === "discount-code") {
+              currentDiscount += parseFloat(billing.paymentMethod.amount);
```

Comment:
This is the line that's throwing an error if `discount.discount` is not parsable as a number.